src/v2/oracles/ChainlinkPriceProvider.sol

```diff
@@ -0,0 +1,84 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.17;
+
+import {
+    AggregatorV3Interface
+} from "@chainlink/interfaces/AggregatorV3Interface.sol";
+import {
+    AggregatorV2V3Interface
+} from "@chainlink/interfaces/AggregatorV2V3Interface.sol";
+import {IVaultFactoryV2} from "../interfaces/IVaultFactoryV2.sol";
+import {IConditionProvider} from "../interfaces/IConditionProvider.sol";
+
+contract ChainlinkPriceProvider is IConditionProvider {
+    uint16 private constant _GRACE_PERIOD_TIME = 3600;
+    uint256 public constant TIME_OUT = 1 days;
```

Comment:
@0xnyxos given that the TIME_OUT window is in correlation to the heartbeat to Chainlinks heartbeat metric, == if price update is older than heartbeat we assume the price is stail and we cant use it to resolve the market, i think that we should pass this variable in the constructor when deploying. As heartbeats can vary and is defined by chainlink